### PR TITLE
Feat/doc build

### PR DESCRIPTION
**Pull request by benoitdm-oslandia**
Add documentation about available flags when building QGIS from command line.

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -342,6 +344,64 @@ ninja pycore
 ninja qgis_desktop
 ```
 
+### 3.7.1 Available compilation flags
+
+QGIS built is tunable according to your needs. Many flags are available to activate or deactivate some functionalities:
```
Comment (nicogodet) [inline INSTALL.md:349]:
```suggestion
QGIS build is tunable according to your needs. Many flags are available to activate or deactivate some functionalities:
```

File: INSTALL.md [hunk 2]
```diff
@@ -342,6 +344,64 @@ ninja pycore
...
+* `WITH_EPT`: Determines whether Entwine Point Cloud (EPT) support should be built
+* `WITH_GRASS${GRASS_CACHE_VERSION}`: Determines whether GRASS ${GRASS_SEARCH_VERSION} plugin should be built
+* `WITH_GUI`: Determines whether QGIS GUI library (and everything built on top of it) should be built
+* `WITH_HANA`: Determines whether SAP HANA Spatial support should be built
+* `WITH_INTERNAL_${ulib}`: Determines whether python ${lib} should be included
+* `WITH_INTERNAL_LAZPERF`: Determines whether LazPerf should be built from internal copy
+* `WITH_INTERNAL_MDAL`: Determines whether MDAL should be built from internal copy
+* `WITH_INTERNAL_POLY2TRI`: Determines whether POLY2TRI should be built from internal copy
+* `WITH_INTERNAL_QWTPOLAR`: Use internal build of QwtPolar
+* `WITH_OAUTH2_PLUGIN`: Build OAuth2 authentication method plugin
```
Comment (nicogodet) [inline INSTALL.md:374]:
```suggestion
* `WITH_OAUTH2_PLUGIN`: Determines whether OAuth2 authentication method plugin should be built
```

Homogenization

File: INSTALL.md [hunk 3]
```diff
@@ -342,6 +344,64 @@ ninja pycore
...
+* `WITH_ORACLE`: Determines whether Oracle support should be built
+* `WITH_PDAL`: Determines whether PDAL support should be built
+* `WITH_POSTGRESQL`: Determines whether POSTGRESQL support should be built
+* `WITH_PY_COMPILE`: Determines whether Python modules in staged or installed locations are byte-compiled
```
Comment (nicogodet) [inline INSTALL.md:378]:
```suggestion
* `WITH_PY_COMPILE`: Determines whether Python modules in staged or installed locations should be byte-compiled
```

Not sure

File: INSTALL.md (hunk 2 above)
Comment (benoitdm-oslandia) [reply INSTALL.md:374]: I am agree but this text is extracted from CMakeLists.txt, should I apply your suggestion?

File: INSTALL.md (hunk 1 above)
Comment (benoitdm-oslandia) [reply INSTALL.md]: done!

File: INSTALL.md [hunk 4]
```diff
@@ -342,6 +344,64 @@ ninja pycore
...
 
+### 3.7.1 Available compilation flags
+
+QGIS build is tunable according to your needs. Many flags are available to activate or deactivate some functionalities:
+
+* `WITH_3D`: Determines whether QGIS 3D library should be built
+* `WITH_ANALYSIS`: Determines whether QGIS analysis library should be built
+* `WITH_APIDOC`: Determines whether the QGIS API doxygen documentation should be built
+* `WITH_ASAN`: Use AddressSanitizer
+* `WITH_ASTYLE`: If you plan to contribute you should reindent with scripts/prepare_commit.sh (using 'our' astyle)
```
Comment (nyalldawson) [inline INSTALL.md:355]: This isn't really necessary anymore -- eg I don't build WITH_ASTYLE enabled. It was needed years ago when distros had older astyle which didn't support some of the code we used, but could likely be safely removed now.

File: INSTALL.md [hunk 5]
```diff
@@ -342,6 +344,64 @@ ninja pycore
...
+* `WITH_AUTH`: Determines whether QGIS authentication methods should be built
+* `WITH_BINDINGS`: Determines whether python bindings should be built
```
Comment (nyalldawson) [inline INSTALL.md:357]:
```suggestion
* `WITH_BINDINGS`: Determines whether Python bindings should be built
```

File: INSTALL.md [hunk 6]
```diff
@@ -342,6 +344,64 @@ ninja pycore
...
+* `WITH_CORE`: Determines whether QGIS core should be built.
+* `WITH_CRASH_HANDLER`: Determines whether the QGIS crash handler application should be built
+* `WITH_CUSTOM_WIDGETS`: Determines whether QGIS custom widgets for Qt Designer should be built
+* `WITH_DESKTOP`: Determines whether QGIS desktop should be built
+* `WITH_EPT`: Determines whether Entwine Point Cloud (EPT) support should be built
+* `WITH_GRASS${GRASS_CACHE_VERSION}`: Determines whether GRASS ${GRASS_SEARCH_VERSION} plugin should be built
+* `WITH_GUI`: Determines whether QGIS GUI library (and everything built on top of it) should be built
+* `WITH_HANA`: Determines whether SAP HANA Spatial support should be built
+* `WITH_INTERNAL_${ulib}`: Determines whether python ${lib} should be included
+* `WITH_INTERNAL_LAZPERF`: Determines whether LazPerf should be built from internal copy
```
Comment (nyalldawson) [inline INSTALL.md:370]:
```suggestion
* `WITH_INTERNAL_LAZPERF`: Determines whether LazPerf should be built from internal copy (recommended)
```

File: INSTALL.md [hunk 7]
```diff
@@ -342,6 +344,64 @@ ninja pycore
...
+* `WITH_INTERNAL_MDAL`: Determines whether MDAL should be built from internal copy
```
Comment (nyalldawson) [inline INSTALL.md:371]:
```suggestion
* `WITH_INTERNAL_MDAL`: Determines whether MDAL should be built from internal copy (recommended)
```

File: INSTALL.md [hunk 8]
```diff
@@ -342,6 +344,64 @@ ninja pycore
...
+* `WITH_OAUTH2_PLUGIN`: Build OAuth2 authentication method plugin
+* `WITH_ORACLE`: Determines whether Oracle support should be built
+* `WITH_PDAL`: Determines whether PDAL support should be built
+* `WITH_POSTGRESQL`: Determines whether POSTGRESQL support should be built
+* `WITH_PY_COMPILE`: Determines whether Python modules in staged or installed locations are byte-compiled
+* `WITH_QGIS_PROCESS`: Determines whether the standalone \"qgis_process\" tool should be built
+* `WITH_QSCIAPI`: Whether to generate PyQGIS QScintilla2 API file. (For devs) run 'make qsci-pap-src' in between QGIS build and install to regenerate .pap file in source tree for console auto-completion.
+* `WITH_QSPATIALITE`: Determines whether QSPATIALITE sql driver should be built
+* `WITH_QT5SERIALPORT`: Determines whether Qt5SerialPort should be tried for GPS positioning
+* `WITH_QT6`: Enable (broken, experimental) Qt6 support
```
Comment (nyalldawson) [inline INSTALL.md:383]:
```suggestion
* `WITH_QT6`: Enable (experimental) Qt6 support
```

it's not broken :)

File: INSTALL.md [hunk 9]
```diff
@@ -342,6 +344,64 @@ ninja pycore
...
+* `WITH_QTWEBKIT`: Enable QtWebkit support
+* `WITH_QUICK`: Determines whether QGIS Quick library should be built
+* `WITH_QWTPOLAR`: Determines whether QwtPolar should be built
```
Comment (nyalldawson) [inline INSTALL.md:386]:
```suggestion
* `WITH_QWTPOLAR`: Determines whether QwtPolar is available and or whether functionality requiring QwtPolar should be disabled
```

File: INSTALL.md (hunk 2 above)
Comment (nicogodet) [reply INSTALL.md:374]: Based on Nyall's review, I would say you are free to change it if you want ! 😀

Comment (benoitdm-oslandia) [reply INSTALL.md:374]: I will modified the INSTALL.md and CMakeLists.txt accordingly

File: INSTALL.md (hunk 3 above)
Comment (benoitdm-oslandia) [reply INSTALL.md:378]: done

File: INSTALL.md (hunk 4 above)
Comment (benoitdm-oslandia) [reply INSTALL.md:355]: done

File: INSTALL.md (hunk 5 above)
Comment (benoitdm-oslandia) [reply INSTALL.md:357]: I have changed all the python by Python

File: INSTALL.md (hunk 7 above)
Comment (benoitdm-oslandia) [reply INSTALL.md:371]: done

File: INSTALL.md (hunk 8 above)
Comment (benoitdm-oslandia) [reply INSTALL.md:383]: done

File: INSTALL.md (hunk 9 above)
Comment (benoitdm-oslandia) [reply INSTALL.md:386]: done

File: INSTALL.md (hunk 6 above)
Comment (benoitdm-oslandia) [reply INSTALL.md]: Does not exist anymore. Bad `find` request.

Comment (benoitdm-oslandia) [reply INSTALL.md]: Bad day today... I did not rebase on the right branch!

**Comment by elpaso** [comment]
Sorry to be the mood breaker here, but I miss the point in adding this information to the documentation, isn't this already available in the CMake file?

Maybe it would be better to add a link to the CMake documentation where is explain to actually access that information.

**Comment by benoitdm-oslandia** [comment]
@elpaso 
>  isn't this already available in the CMake file?

are you talking about the `CMakeList.txt` file? IMHP I do not think any new dev will be reading the `CMakeList.txt` file to understand what are the available flags and what they do. 

But if you know a CMake command to display all the available flags, it could replace my piece of documentation.

**Comment by benoitdm-oslandia** [comment]
@elpaso @nicogodet? Can you review the last changes please?

**Comment by elpaso** [comment]
> IMHP I do not think any new dev will be reading the `CMakeList.txt` file to understand what are the available flags and what they do.

Any decent IDE and the recommended one (QtCreator) has a way to see what's in the cmake config file:

immagine


But, if you commit yourself to keep this information up to date for the foreseeable future, I'm ok with that.

**Comment by benoitdm-oslandia** [comment]
There is, now, only a subset of these flags and I provide the way to retrieve the complete list on the command line (as I often do not have an IDE)

**Review by elpaso — APPROVED** [review]

**Comment by benoitdm-oslandia** [comment]
thanks @elpaso

**Comment by elpaso** [comment]
> thanks @elpaso

Thank **you**, and sorry for my criticism :)

I am only trying to avoid adding information that might become obsolete and stay there forever, we have too much of that already.

**Pull request merged**